docs/deep-dive-blog-posts.md

```diff
@@ -30,3 +30,4 @@
 - [Preventing .NET Garbage Collections with the TryStartNoGCRegion API](http://mattwarren.org/2016/08/16/Preventing-dotNET-Garbage-Collections-with-the-TryStartNoGCRegion-API/)
 - [GC Pauses and Safe Points](http://mattwarren.org/2016/08/08/GC-Pauses-and-Safe-Points/)
 - [Strings and the CLR - a Special Relationship](http://mattwarren.org/2016/05/31/Strings-and-the-CLR-a-Special-Relationship/)
+- [How Async/Await Really Works in C#](https://devblogs.microsoft.com/dotnet/how-async-await-really-works/)
```

Comment:
The other posts in `Posts that reference specific parts of the source` section looks to be in newest first order. Can look a little inconsistent.

Reply:
Thanks. I don't think time order matters here, especially since this is unrelated to the other topics. We might want to put in place some other structure, though.